kasa/smart/modules/powerprotection.py

```diff
@@ -0,0 +1,105 @@
+"""Power protection module."""
+
+from __future__ import annotations
+
+from ...feature import Feature
+from ..smartmodule import SmartModule
+
+
+class PowerProtection(SmartModule):
+    """Implementation for power_protection."""
+
+    REQUIRED_COMPONENT = "power_protection"
+
+    def _initialize_features(self) -> None:
+        """Initialize features after the initial update."""
+        self._add_feature(
+            Feature(
+                device=self._device,
+                id="overloaded",
+                name="Overloaded",
+                container=self,
+                attribute_getter="overloaded",
+                type=Feature.Type.BinarySensor,
+                category=Feature.Category.Info,
+            )
+        )
+        self._add_feature(
+            Feature(
+                device=self._device,
+                id="power_protection_enabled",
+                name="Power protection enabled",
+                container=self,
+                attribute_getter="enabled",
+                attribute_setter="set_enabled",
+                type=Feature.Type.Switch,
+                category=Feature.Category.Config,
+            )
+        )
+        self._add_feature(
+            Feature(
+                device=self._device,
+                id="power_protection_threshold",
+                name="Power protection threshold",
+                container=self,
+                attribute_getter="protection_threshold",
+                attribute_setter="set_protection_threshold",
+                unit_getter=lambda: "W",
+                type=Feature.Type.Number,
+                range_getter="protection_threshold_range",
+                category=Feature.Category.Config,
+            )
+        )
+
+    def query(self) -> dict:
+        """Query to execute during the update cycle."""
+        return {"get_protection_power": None, "get_max_power": None}
```

Comment:
Are you able to test this on your own devices? I'm wondering if we should start favouring `{}` instead of `None` for empty params? Can you see what the app is sending for these?

Reply:
Will change that after testing it, good catch!